netbox_data_import/templates/netbox_data_import/import_preview.html

```
@@ -1002,6 +1213,55 @@ <h5 class="modal-title"><i class="mdi mdi-plus-circle"></i> Add Class Mapping</h
+            {% if can_create_role %}
+            <div class="mt-2">
+              <button type="button" class="btn btn-sm btn-outline-primary" id="cm_create_role_toggle"
+                      onclick="cmToggleCreateRole()">
+                <i class="mdi mdi-plus"></i> Create new role
+              </button>
+              <div id="cm_create_role_form" style="display:none;" class="border rounded p-3 mt-2 bg-body-secondary">
+                <p class="text-muted small mb-2">Create a new DeviceRole directly in NetBox and use it here.</p>
+                <div class="row g-2 mb-2">
+                  <div class="col-7">
+                    <label class="form-label small mb-1">Role name</label>
+                    <input type="text" id="cm_new_role_name" class="form-control form-control-sm"
```

HTMLHint (1.9.2)

[warning] No matching [ label ] tag found.

(input-requires-label)